src/rte/RTESolver.jl

```diff
@@ -13,28 +13,81 @@ using ..RTE
 using ..Optics
 using ..LookUpTables
 
-export solve_lw!, solve_sw!
+export solve_lw!, solve_sw!, apply_metric_scaling!
 
 include("longwave1scalar.jl")
 include("longwave2stream.jl")
 include("shortwave1scalar.jl")
 include("shortwave2stream.jl")
 
 """
-    solve_lw!((; context, flux, src, bcs, op)::NoScatLWRTE, as::GrayAtmosphericState)
+    apply_metric_scaling!(flux::FluxLW, metric_scaling)
 
-Non-scattering RTE solver for the longwave problem, using gray optics.
+Apply metric scaling factor for longwave radiative fluxes. This accounts for geometric expansion of 
+grid columns with increasing altitude when deep-atmosphere metric terms are used. Requires
+`flux_up`, `flux_dn` and `flux_net` to be available in `flux`. The metric scaling factor is 
+an optional argument that has a scalar value of `1` when shallow atmosphere approximations
+are used (in which case this function returns nothing).
 """
-solve_lw!((; context, flux, src, bcs, op, angle_disc)::NoScatLWRTE, as::GrayAtmosphericState) =
+function apply_metric_scaling!(flux::FluxLW, metric_scaling::M) where {M <: AbstractArray}
+    flux.flux_up .= flux.flux_up ./ metric_scaling
+    flux.flux_dn .= flux.flux_dn ./ metric_scaling
+    flux.flux_net .= flux.flux_net ./ metric_scaling
+end
+function apply_metric_scaling!(flux::FluxLW, metric_scaling::M) where {M <: Union{AbstractFloat, Int}}
+    nothing
+end
```

Comment:
```suggestion
apply_metric_scaling!(flux, metric_scaling::Nothing) = nothing
```
The use of `Union{AbstractFloat, Int}` is somewhat confusing here, as I would expect `apply_metric_scaling!(flux, 2)` to multiply the fluxes by 2 instead of doing nothing. It would be better to have the default value of `metric_scaling` be `nothing` instead of 1, so that there is no ambiguity.